gymnasium/spaces/discrete.py

```diff
@@ -21,9 +21,9 @@ class Discrete(Space[int]):
 
     def __init__(
         self,
-        n: int,
+        n: int | np.int64,
```

Comment:
Can't we use one of the general numpy dtypes to allow int/int64/int32/whatever other int? The annotation could be `int | np.integer`, then I think it would accept `int`, `int64`, `int32`, `uint8` and all of that stuff. While it does make sense to narrow down the return type, this would throw redundant errors (since all of those can be cast to `int64` anyways)

Reply:
Also in L24

Reply:
Done